docs/src/main/paradox/scala/http/server-side/http2.md

```diff
@@ -0,0 +1,146 @@
+curl 7.52.1 (x86_64-pc-linux-gnu) libcurl/7.52.1 OpenSSL/1.0.2l zlib/1.2.8 libidn2/0.16 libpsl/0.17.0 (+libidn2/0.16) libssh2/1.8.0 nghttp2/1.23.1 librtmp/2.3
+Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtmp rtsp scp sftp smb smbs smtp smtps telnet tftp
+Features: AsynchDNS IDN IPv6 Largefile GSS-API Kerberos SPNEGO NTLM NTLM_WB SSL libz TLS-SRP HTTP2 UnixSockets HTTPS-proxy PSL
+```
+
+When you connect to your service you should now see:
+
+```
+$ curl -k -v https://localhost:8443
+(...)
+* ALPN, offering h2
+* ALPN, offering http/1.1
+(...)
+* ALPN, server did not agree to a protocol
+(...)
+> GET / HTTP/1.1
+(...)
+```
+
+This shows `curl` declaring it is ready to speak either `h2` (the shorthand name of HTTP/2) or HTTP/1.1, but because it could not determine whether the server is ready to it fell back to HTTP/1.1.
+
+## Application-Layer Protocol Negotiation (ALPN)
+
+[Application-Layer Protocol Negotiation (ALPN)](https://en.wikipedia.org/wiki/Application-Layer_Protocol_Negotiation) is used to negotiate whether both client and server support HTTP/2.
+
+ALPN support comes with the JVM starting from version 9. If you're on a previous version of the JVM, you'll have to load a Java Agent to provide this functionality.
+
+## Run and package
+
+### sbt
+
+sbt can be configured to load the agent with the [sbt-javaagent plugin](https://github.com/sbt/sbt-javaagent):
+
+```
+  .enablePlugins(JavaAgent)
+  .settings(
+    javaAgents += "org.mortbay.jetty.alpn" % "jetty-alpn-agent" % "2.0.6" % "runtime"
+  )
+```
+
+This should automatically load the agent when running, testing, or even in distributions made with [sbt-native-package](https://github.com/sbt/sbt-native-packager).
+
+@@@ div { .group-java}
+
+### maven
+
+To configure maven to load the agent when running `mvn exec:exec`, add it as a 'runtime' dependency:
+
+```
+<dependency>
+    <groupId>org.mortbay.jetty.alpn</groupId>
+    <artifactId>jetty-alpn-agent</artifactId>
+    <version>2.0.6</version>
+    <scope>runtime</scope>
+</dependency>
+```
+
+and use the `maven-dependency-plugin`:
+
+```
+<plugin>
+    <artifactId>maven-dependency-plugin</artifactId>
+    <version>2.5.1</version>
+    <executions>
+        <execution>
+            <id>getClasspathFilenames</id>
+            <goals>
+                <goal>properties</goal>
+            </goals>
+        </execution>
+     </executions>
+</plugin>
+```
+
+to add it to the `exec-maven-plugin` arguments:
+
+```
+<plugin>
+    <groupId>org.codehaus.mojo</groupId>
+    <artifactId>exec-maven-plugin</artifactId>
+    <version>1.6.0</version>
+    <configuration>
+        <executable>java</executable>
+        <arguments>
+            <argument>-javaagent:${org.mortbay.jetty.alpn:jetty-alpn-agent:jar}</argument>
+            <argument>-classpath</argument>
+            <classpath />
+            <argument>com.example.HttpServer</argument>
+        </arguments>
+    </configuration>
+</plugin>
+```
+
+@@@
```

Comment:
Also show how to do it from a raw jar with the "app.jar" and the "agent.jar", it's easiest to understand what's going on and easiest to translate to some automation tool I think...